src/file.toit

```diff
@@ -251,9 +259,83 @@ Deletes a file, given its name.
 delete name/string -> none:
   #primitive.file.unlink
 
-/*
+/**
 Renames a file or directory.
 Only works if the $to name is on the same filesystem.
 */
 rename from/string to/string -> none:
   #primitive.file.rename
+
+/**
+Creates a hard link from $source to $target.
+*/
+link --hard --source/string --target/string -> none:
+  if not hard: throw "INVALID_ARGUMENT"
+  link_ source target LINK_TYPE_HARD_
+
+/**
+Creates a soft link from $source to $target.
+  Note: On Posix systems this works for both directories and files.
+  On Windows systems this works only for file links.
+*/
+link --soft --source/string --target/string -> none:
+  if not soft: throw "INVALID_ARGUMENT"
+  link_ source target LINK_TYPE_SYMBOLIC_
+
+/**
+Creates a soft directory link from $source to $target. (For the windows platform).
+*/
+link --soft-dir --source/string --target/string -> none:
+  if not soft-dir or system.platform != system.PLATFORM-WINDOWS: throw "INVALID_ARGUMENT"
```

Comment:
For the case where you want to write portable code, but the target does not exist, it would be nice if this just worked on Linux too.  Please rename the flag to soft-directory, since we try to avoid abbreviations unless they are more common than the unabbreviated name.